### PR TITLE
feat: add integration tests for incident client api

### PR DESCRIPTION
**Pull request by ralfpuchert**
## Description

* Add 'get incident' client api
* Add integration tests for incident client api

## Related issues

Related to #23365

### CODE REVIEW

File: qa/integration-tests/src/test/java/io/camunda/it/client/IncidentQueryTest.java [hunk 1]
```diff
@@ -0,0 +1,514 @@
...
+  // TODO:
+  // - processDefinitionKey
+  // - processDefinitionId
+  // - errorType
+  // - errorMessage
+  // - flowNodeId
+  // - creationTime
+  // - jobKey
+  // - treePath
+  // - tenantId
```
Comment (tmetzke) [inline qa/integration-tests/src/test/java/io/camunda/it/client/IncidentQueryTest.java:146]: 🙃 Looks to me like you added those, but please double-check again

File: qa/integration-tests/src/test/java/io/camunda/it/client/IncidentQueryTest.java [hunk 2]
```diff
@@ -0,0 +1,514 @@
...
+    final var sortedDesc = all.stream().sorted(Comparator.reverseOrder()).toList();
+
+    assertThat(resultAsc.items().stream().map(Incident::getCreationTime).toList())
+        .containsExactlyElementsOf(sortedAsc);
+    assertThat(resultDesc.items().stream().map(Incident::getCreationTime).toList())
+        .containsExactlyElementsOf(sortedDesc);
+  }
+
+  // Implement the following tests for the remaining fields: type, processDefinitionKey,
+  // processInstanceKey, tenantId, flowNodeInstanceId, flowNodeId, state, jobKey, creationTime
```
Comment (tmetzke) [inline qa/integration-tests/src/test/java/io/camunda/it/client/IncidentQueryTest.java:453]: 🙃 As far as I can see, this is done

**Review by tmetzke — APPROVED** [review]
👍 Looks good to me, great improvement 🚀

File: qa/integration-tests/src/test/java/io/camunda/it/client/IncidentQueryTest.java (hunk 1 above)
Comment (ralfpuchert) [reply qa/integration-tests/src/test/java/io/camunda/it/client/IncidentQueryTest.java:146]: Yes, came from GitHub Copilot. Need to clean this up. 🙈

File: qa/integration-tests/src/test/java/io/camunda/it/client/IncidentQueryTest.java (hunk 2 above)
Comment (ralfpuchert) [reply qa/integration-tests/src/test/java/io/camunda/it/client/IncidentQueryTest.java:453]: Same here. Thanks.

**Comment by cmur2** [comment]
One of the new tests apparently leads to flakiness in the CI in the merge queue and fails sporadically, see [discussion on Slack](https://camunda.slack.com/archives/C071KP5BTHB/p1729173143391669). Please prioritize mitigating the problem here e.g. via revert.

**Comment by cmur2** [comment]
@ralfpuchert @tmetzke ^

**Pull request merged**